### PR TITLE
rsweb-7370-zoolander-documentation

### PR DESCRIPTION
**Pull request by amberRrucker**
RSWEB-7370
feat(docs):adding and improving documentation

This would include:
-Remove unused variables
-Add updated images of examples
-Removing outdated examples (ticker, old banner style)
-Ensuring we update the landing page examples and documentation
-Make sure all examples are properly documented 
-Cleaning up the UI of the examples page
-Add documentation to zoolander solutions usage that details when using the same patterns keep it on the same background/style
-Add documentation around required fields
-Add documentation that all pages should begin with a lead that is not included within a river.
-Add documentation for disc type lists

### CODE REVIEW

File: styleguide/derek/assets/index.jade [hunk 1]
```diff
@@ -14,9 +14,15 @@ block page
...
           li
+            a(href="derek/assets/swatches") Colors &amp; Swatches
+          li
             a(href="derek/assets/icon-font") Icon Font
           li
             a(href="derek/assets/photography") Photography &amp; Banners
+          li
+            a(href="derek/assets/tables") Tables
+          li
+            a(href="derek/assets/tables") Forms
```
Comment (WidgetsBurritos) [inline styleguide/derek/assets/index.jade]: Incorrect link here. Also if we're gonna link to the form styles, we probably should remove them from the asset page itself:
https://github.com/amberRrucker/zoolander/blob/c2f20da020cb28fa9661e9de89eb6a3a84b72428/styleguide/derek/assets/index.jade#L79

File: styleguide/_layouts/default.jade [hunk 2]
```diff
@@ -31,50 +31,48 @@ html(lang="en")
...
                 a(href="derek/assets/photography").navZoolander-dropdownLink Photography/Banners
               li.navZoolander-dropdownItem
-                a(href="derek/assets/icon-font").navZoolander-dropdownLink Tabs
+                a(href="derek/assets/swatches").navZoolander-dropdownLink Colors/Swatches
+              li.navZoolander-dropdownItem
+                a(href="derek/assets/icon-font").navZoolander-dropdownLink Icon Font
+              li.navZoolander-dropdownItem
+                a(href="derek/assets/buttons").navZoolander-dropdownLink Buttons/CTAs
               li.navZoolander-dropdownItem
                 a(href="derek/examples/tables").navZoolander-dropdownLink Tables
```
Comment (WidgetsBurritos) [inline styleguide/_layouts/default.jade]: This link needs to change to `derek/assets/tables`

File: styleguide/derek/index.jade [hunk 3]
```diff
@@ -25,11 +25,15 @@ block page
...
         ul.nav.nav-pills.nav-stacked
           li
-            a(href="derek/examples") Examples
+            a(href="derek/solutions") Solutions
+          li
+            a(href="derek/templates") Templates
           li
             a(href="derek/development") Development
           li
             a(href="derek/workflow") Workflow/Process
```
Comment (WidgetsBurritos) [inline styleguide/derek/index.jade]: For these incomplete pages, should we have a `(Coming Soon)` or maybe just hide them for now, as the links to "Need to Add" pages are kinda annoying.

File: styleguide/_layouts/default.jade [hunk 4]
```diff
@@ -31,50 +31,48 @@ html(lang="en")
...
+          //Add in themes dropdown when there are multiple
 
           li.navZoolander-item
             a(href="#").navZoolander-link.navZoolander-hasDropdown Derek Theme
             ul.navZoolander-dropdown
               li.navZoolander-item
                 a(href="derek/").navZoolander-dropdownLink Overview
               li.navZoolander-item
-                a(href="derek/examples").navZoolander-dropdownLink Examples
+                a(href="derek/solutions").navZoolander-dropdownLink Solutions
```
Comment (WidgetsBurritos) [inline styleguide/_layouts/default.jade]: By moving everything out of /examples/ we no longer have the auto-collapse navigation feature.  Please adjust the JS accordingly. 

https://github.com/rackerlabs/zoolander/blob/bcdb78584d9a2a614c13e33b773846f0880cf38e/styleguide/js/zoolander.js#L184

File: styleguide/derek/solutions/solutions.ejs [hunk 5]
```diff
@@ -17,9 +17,9 @@
 </div>
```
Comment (WidgetsBurritos) [inline styleguide/derek/solutions/solutions.ejs]: Please update the link above to point to the new solutions-usage URL. 
https://github.com/rackerlabs/zoolander/pull/293/commits/c2f20da020cb28fa9661e9de89eb6a3a84b72428#diff-8a37516e22a27a627a72eb50bfff1809R5

File: styleguide/_layouts/default.jade [hunk 6]
```diff
@@ -31,50 +31,48 @@ html(lang="en")
...
-                a(href="derek/assets/forms").navZoolander-dropdownLink Forms
-              li.navZoolander-dropdownItem
-                a(href="derek/assets/buttons").navZoolander-dropdownLink Buttons/CTAs
-              li.navZoolander-dropdownItem
                 a(href="derek/assets/typography").navZoolander-dropdownLink Typography
               li.navZoolander-dropdownItem
                 a(href="derek/assets/photography").navZoolander-dropdownLink Photography/Banners
               li.navZoolander-dropdownItem
-                a(href="derek/assets/icon-font").navZoolander-dropdownLink Tabs
+                a(href="derek/assets/swatches").navZoolander-dropdownLink Colors/Swatches
```
Comment (WidgetsBurritos) [inline styleguide/_layouts/default.jade]: We probably shouldn't worry about it in this PR, but for future PR purposes:

Does it make sense to have Global examples under the `/derek/` URL path, if there's a separate set of navigation for the Derek Theme itself.  Should they be under `/global/` instead? That has potential to cause confusion.  Once again, not a question to get addressed in this PR, but worth thinking about.

File: styleguide/derek/incubation/events.ejs [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+              <div class="eventsOverview-weekEvent">
+                <div class="eventsOverview-weekEvent-header">
+                  <div class="eventsOverview-weekEvent-headerTitle">Rackspace::Solve San Francisco</div>
+                  <div class="eventsOverview-weekEvent-headerLocation">San Francisco, CA</div>
+                </div>
+                <div class="eventsOverview-weekEvent-date">September <%= i * 7 + 1 + j %>, 2016</div>
+                <div class="eventsOverview-weekEvent-details">
+                  Andouille corned beef pig doner. Jowl frankfurter ball tip flank short loin short ribs. Tri-tip prosciutto leberkas capicola pork belly fatback meatball, biltong venison jerky hamburger picanha. Corned beef turducken tongue ribeye, shankle salami boudin. Boudin doner shankle t-bone ball tip short loin tongue short ribs ham.
+                </div>
+                <a class="eventsOverview-weekEvent-learnMore" href="/derek/examples/events-interior">Learn More</a>
```
Comment (WidgetsBurritos) [inline styleguide/derek/incubation/events.ejs]: Link needs to be adjusted

File: styleguide/derek/incubation/index.jade [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+      h5 Homepage
+      a(href="derek/examples/homepage")
+        img.img-thumbnail(src="imgs/derek/examples/homepage.png")
+    .col-md-4
+      h5 Event Overview
+      a(href="derek/examples/events")
+        img.img-thumbnail(src="imgs/derek/examples/event-overview.png")
+    .col-md-4
+      h5 Story Overview
+        a(href="derek/examples/stories")
```
Comment (WidgetsBurritos) [inline styleguide/derek/incubation/index.jade]: All of the links on this page no longer work.

**Comment by amberRrucker** [comment]
Made all requested changes and double checked all hrefs. Do we want to split commits? How do we want to split them?

File: styleguide/js/zoolander.js [hunk 9]
```diff
@@ -181,6 +181,6 @@ zoolanderSlideBtn.unbind().click(function(e){
 
 //Zoolander only solution for auto collapsing menu on example pages
 var url = window.location.pathname;
-if(url.match(/\/derek\/examples\//gi) && url != '/derek/examples/' && url != '/derek/examples/landing-page/'){
+if(url.match(/\/derek\/incubation\//gi) && url != '/derek/incubation/'){
```
Comment (WidgetsBurritos) [inline styleguide/js/zoolander.js]: We also need `derek/solutions/solutions` included

File: styleguide/derek/incubation/events-interior.ejs [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+        </div>
+        <div class="event-details">
+          <div class="event-detailsTitle">Details</div>
+          <p>Join us at Rackspace::Solve San Francisco, a free conference where you’ll learn how the latest cloud solutions can help you scale your business, get your products to market quickly, and reduce the cost of innovation.</p>
+          <p>Sessions cover real world solutions to cloud security, big data, and private cloud challenges. Visit our Solutions Pavilion to discuss your company’s specific needs and whiteboard solutions with Rackspace experts and partners.</p>
+          <p>Beyond empowering companies to tap into the power of the cloud, Rackspace also strives for social empowerment. In that spirit, we’ll make a monetary donation of $15 per attendee to Girls Who Code – up to $5,000. In addition to a great day of learning and networking, your attendance at Rackspace::Solve will help give high school girls the skills and resources they need to pursue opportunities in computing fields.</p>
+        </div>
+      </div>
+      <div class="event-row">
+        <a class="event-backToEvents" href="/derek/examples/events">Back to upcoming events</a>
```
Comment (WidgetsBurritos) [inline styleguide/derek/incubation/events-interior.ejs]: this path still needs to change

**Comment by WidgetsBurritos** [comment]
Oh and also, you'll need a rebase on this.

**Comment by WidgetsBurritos** [comment]
Needs another rebase.

**Comment by WidgetsBurritos** [comment]
👍

**Comment by brantdarilek** [comment]
@amberRrucker 
I found an issue with the tooltips on the swatches page. And the list partials are not getting rendered on the typography page. You should be able to see the issues with my review comments.

**Comment by amberRrucker** [comment]
Hey @brantdarilek, These are known issues that arent in the scope of this task. I'm going to need to write these into the next task, since the scope of this one is already so large and we're trying to get these documentation tasks a little less intense these days.
@WidgetsBurritos is that good with you as well? Trying to slim down on scope creep with zoolander-isms.

**Comment by brantdarilek** [comment]
@amberRrucker @WidgetsBurritos 
That sounds good to me. 👍

**Comment by WidgetsBurritos** [comment]
yeah let's not get hung up on that. There is some css removal here though so we may need to do some QE with this task.

**Comment by SyntaxStacks** [comment]
screen shot 2016-11-17 at 3 19 05 pm

Please take a look at the home page and office 365 page in /derek/incubation 
things do not look right

**Comment by WidgetsBurritos** [comment]
@SyntaxStacks whoa, I think those issues are already in zoolander.  Have no idea when they came in, but that's fun!

**Comment by amberRrucker** [comment]
@widgetsburritos I fixed it! Haha. So pretty.

**Comment by WidgetsBurritos** [comment]
@amberRrucker what was it?

**Comment by amberRrucker** [comment]
@widgetsburritos it was some extra code in the homepage. Probably just a weird merge.

**Comment by SyntaxStacks** [comment]
👍

**Pull request merged**